Improve the readability of 'getShippingDetails()' and 'updateDetails' method of UserShippingDetails component

adityasharma7 opened the pull request:
closes #3770

### Related issues
<!--  Put related issue number which this PR is closing. For example #123 -->

closes #

### Short description and why it's useful
<!-- describe in a few words what is this Pull Request changing and why it's useful -->



### Screenshots of visual changes before/after (if there are any)
<!-- if you made any changes in the UI layer please provide before/after screenshots -->

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [x] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [x] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance
<!-- Please get familiar with following info -->

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

andrzejewsky commented on core/modules/user/components/UserShippingDetails.ts:
```diff
@@ -148,53 +148,40 @@ export const UserShippingDetails = {
+              shippingDetails.country = address.country_id
+              shippingDetails.phone = address.hasOwnProperty('telephone') ? address.telephone : ''
+              // Stop iteration as default_shipping address is found
+              break;
```
instead of doing `breaks`, please move this to the separate method or even to the helper, eg:

```
readShippingDetailsFromCurrentUser() {
  for (let address of this.currentUser.addresses) {
     if (toString(address.id) === toString(this.currentUser.default_shipping)) {
        return { 
           firstName: address.firstname,
           lastName: address.lastname,
           street: address.street[0],
           house: address.street[1],
           .....
        }
     }
  }
  return {}
}
```

then you can use simply:
`shippingDetails = Object.assign(shippingDetails, readShippingDetailsFromCurrentUser())`

adityasharma7 replied:
Makes sense. Thanks for the review @andrzejewsky. I observed similar code pattern in some other code blocks like updatedShippingDetails(), fillCompanyAddress(). I will update the PR soon with changing those blocks using the new method.

adityasharma7 commented:
@andrzejewsky @pkarw I have updated the PR with changes suggested by @andrzejewsky. Apart from that, I have also simplified the code for updateDetails() method along with the update to the changelogs.

andrzejewsky commented:
@adityasharma7 thanks! @alinadivante please have look at checkout process, request data for shipping details and whether entire placing a order works as beforehand, thanks!
